src/zarr.hh

```diff
@@ -1,41 +1,27 @@
 #ifndef H_ACQUIRE_STORAGE_ZARR_V0
 #define H_ACQUIRE_STORAGE_ZARR_V0
 
+#ifndef __cplusplus
+#error "This header requires C++20"
+#endif
+
```

Comment:
Optional

Given that the implementation is quite a bit more complex now (so that we can easily accommodate v2 and v3 writers), I'd really value a short description of the key components of the implementation and how they're related - really just a list of class names and one line summaries.

This seems like a good place for this (i.e. it's one of the first places I would look) or alternatively a `README` in `src.

Reply:
I added a README here.